bin/p2-watch/main.go

```diff
@@ -122,3 +136,56 @@ func watchPodClusters(client consulutil.ConsulClient, applicator labels.Applicat
 		log.Fatalf("error watching pod cluster: %v", err)
 	}
 }
+
+func watchHealth(service string, client consulutil.ConsulClient) {
+	hc := checker.NewConsulHealthChecker(client)
+	healthResults := make(chan map[types.NodeName]health.Result)
+	errCh := make(chan error)
+	quitCh := make(chan struct{})
+	go hc.WatchService(*healthService, healthResults, errCh, quitCh)
+
+	go func() {
+		signalCh := make(chan os.Signal, 2)
+		signal.Notify(signalCh, syscall.SIGTERM, os.Interrupt)
+		received := <-signalCh
+		log.Printf("[WARN] Received %v, shutting down", received)
+		close(quitCh)
+	}()
+
+	var hr map[types.NodeName]health.Result
+	var err error
+	for {
+		select {
+		case hr = <-healthResults:
+			var sortedHealthResults nodeHealthResults
+			for _, r := range hr {
+				sortedHealthResults = append(sortedHealthResults, r)
+			}
+			sort.Sort(sortedHealthResults)
+			for _, r := range sortedHealthResults {
+				fmt.Printf("%s %s\n", r.Node.String(), r.Status)
+			}
+			fmt.Printf("\n")
+		case err = <-errCh:
+			log.Printf("[ERROR] %v\n", err)
+		case <-quitCh:
+			os.Exit(0)
+		default:
+			time.Sleep(200 * time.Millisecond)
+		}
+	}
+}
+
+type nodeHealthResults []health.Result
+
+func (hrs nodeHealthResults) Len() int {
+	return len(hrs)
+}
+
+func (hrs nodeHealthResults) Less(i, j int) bool {
+	return hrs[i].Node.String() < hrs[j].Node.String()
```

Comment:
I don't know what other ordering you had in mind. I do like having the nodes in a stable order, so I think node name seems good. Were you thinking of something else?